### DISCUSSION TITLE
Deleted file from external library still showing in Immich

### DISCUSSION THREAD

**Discussion by Gitonthegrid** [immich-app/immich#7529] [General]
Hi all, 

I am wondering how the external library works. 
I have setup a external test folder on my NAS, and added the folder to Immich external folder settings in de settings. 

This all is working fine, and the images that I put in that folder, are showing up in Immich. So far so good. 
The only problem is when I delete images from that folder. Immich is the still showing the thumbnail and I cant get rid of it. 
I cant open the original anymore, so Immich knows the file is deleted, but I cant do anything to get rid of the thumbnails. 

I what kind a way is Immich updating files in that external folder?

**Comment by schuhbacca** [comment]
![image](https://github.com/immich-app/immich/assets/5739463/89b6ac92-8a87-4678-91ba-0a2f7ac4dfca)

**Reply by hulmgulm** [reply]
With Immich version 1.116.0, the "Remove offline files" context menu entry for external libraries vanished. What happened to it? It looks like a scan does not remove offline files for me.

**Reply by hulmgulm** [reply]
The "Remove offline files" feature got reworked in this PR https://github.com/immich-app/immich/pull/12220. It works differently now, which is fine for me. I somehow missed it when looking over the release notes for 1.116.0.

**Comment by Gitonthegrid** [comment]
@schuhbacca 
I am sorry I didn't mention. But I already did this, and it say "removing offline files" But the files (thumbnails) are still there.

**Reply by schuhbacca** [reply]
Did you do a full library scan before remove offline files?

**Comment by jenda69** [comment]
Immich has to notice that the file is deleted first. In my setup, it usually takes a couple of hours until "Asset is offline" notification displays in Info for deleted picture. Then it can be deleted using "Remove offline files".
I have no idea, how to make the process faster. Refreshing metadata doesn't add the offline flag and I think even scanning library didn't do that when I tried.

**Reply by Gitonthegrid** [reply]
Even though I completely removed the containers and removed the external library, all the pictures are still showing. 
So there is a really anoying bug. I really want to start fresh. But even with a complete remove off the containers and remove the volume setting, it still showing all pictures when I redeploy the stack. 

I cant do anything now. Is I really want to force remove all data from the external library (within immich)

**Reply by wildleaf** [reply]
Did you check the container logs and see any errors? I've experience something similar earlier. It failed to insert audit logs due to duplicate id violation. It worked after I deleted everything in the audit table.

**Reply by aisbergde** [reply]
I don't think that this is a bug, but it is a feature. The issue is, that the feature is hard to understand and this leads to be misinterpreted. And I also misunderstood it at the beginning.

immich is not showing assets, but the generated thumbnails. If it's an image and you don't need to download, then the original asset isn't required in most cases.

That's why all assets are still showing even when no more existing. And (in my understanding) there are some daily maintenance tasks to clean up. That's why you have to wait some hours. The daily maintenance can be forced by "rescan all libraries". If I wrong, someone could correct me.

**Comment by SayantanRC** [comment]
I did the following and my deleted files seem to have gone.
1. Re-scan all library files
2. Remove offline files

Immich version 1.111.0

**Reply by leet1994** [reply]
> I did the following and my deleted files seem to have gone.
> 
> 1. Re-scan all library files
> 2. Remove offline files
> 
> Immich version 1.111.0

Do you happen to know if this can be automated, i.e. as a cronjob?

**Reply by SayantanRC** [reply]
No, I only need to do it very rarely.

**Comment by hulmgulm** [comment]
With Immich version 1.116.0, the "Remove offline files" context menu entry for external libraries vanished. What happened to it? It looks like a scan does not remove offline files for me.

**Reply by stilsch** [reply]
Same for me. I found no way to get rid of deleted files.
